gson/src/main/java/com/google/gson/stream/JsonReader.java

```diff
@@ -222,6 +222,9 @@ public class JsonReader implements Closeable {
   private static final int NUMBER_CHAR_EXP_E = 5;
   private static final int NUMBER_CHAR_EXP_SIGN = 6;
   private static final int NUMBER_CHAR_EXP_DIGIT = 7;
+  private static final int MAX_STACK_SIZE = 32;
```

Comment:
This is not the max stack size; it is the initial stack size, see `JsonReader.push`. I am also not sure if this value needs a dedicated `static final` field.